ansible/roles/test/tasks/neighbour-mac-noptf.yml

```diff
@@ -48,15 +48,20 @@
     set_fact:
       v4_intf: "Ethernet0"
       v4_nei: "{{ v4_intf_nei }}"
+      need_add_ip: True
     when: v4_nei is not defined
 
+  - name: startup Ethernet0
+    command: "config interface startup Ethernet0"
+    when: need_add_ip is defined and need_add_ip == True
+
   - name: add an ip entry for Ethernet0
-    command: "/sbin/ifconfig Ethernet0 {{ v4_intf_ip }}"
-    when: v4_nei is not defined
+    command: "config interface ip add Ethernet0 {{ v4_intf_ip }}"
+    when: need_add_ip is defined and need_add_ip == True
 
   - name: add neighbor of Ethernet0
     command: "/sbin/ip neigh add {{ v4_intf_nei }} lladdr {{ v4_mac1 }} dev Ethernet0"
-    when: v4_nei is not defined
+    when: need_add_ip is defined and need_add_ip == True
```

Comment:
You can consider putting the touched tasks in a block with condition "when: v4_nei is not defined". Then the extra variable "need_add_ip" can be removed and the logic could be simpler.

Reply:
fixed